bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/items/DimmerItem.java

```diff
@@ -79,6 +79,8 @@ public void setState(State state) {
             super.setState(PercentType.ZERO);
         } else if (state == OnOffType.ON) {
             super.setState(PercentType.HUNDRED);
+        } else if (state.getClass() == DecimalType.class) {
+            super.setState(new PercentType(((DecimalType) state).toBigDecimal().multiply(new BigDecimal(100))));

```

Comment:
When setting the state, do a conversion from DecimalType to PercentType, what's interesting is that on line 101 below the class was already doing a conversion the other way when getting the state.